src/pages/Post/Post.tsx

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable prettier/prettier */
+import './Post.css'
+
+import axios from 'axios'
+import React, {useEffect, useState } from 'react'
+
+interface IPost {
+    id: number,
+    links: {id: number, caption: string, url: string}[],
+    caption: string, 
+    short_text: string, 
+    details: string,
+    added_at: string,
+    show_after: string,
+    disable_after: string
+}
+
+const defaultPosts: IPost[] = []
+
+export const Posts: React.FC = () => {
+    const [posts, setPosts]: [IPost[], (posts: IPost[]) => void] = useState(defaultPosts)
+    const [loading, setLoading]: [boolean, (loading: boolean) => void] = React.useState<boolean>(true)
+    const [error, setError]: [string, (error: string) => void] = React.useState("")

```

Comment:
niekedy to netreba s typmi prehanat, typescript si vie vela typov inferovat (odvodit) z default hodnotky. staci dokonca len takto:
```suggestion
  const [posts, setPosts] = useState(defaultPosts)
  const [loading, setLoading] = useState(true)
  const [error, setError] = useState('')
```